lib/Jsrt/ChakraDebug.h

```diff
@@ -636,87 +637,93 @@ typedef UINT32 DWORD;
-    /// <param name="fullTTDUri">The fully resolved location for the TTD data output.</param>
-    typedef void (CHAKRA_CALLBACK *JsTTDInitializeUriCallback)(_In_z_ const wchar_t* uri, _Out_ wchar_t** fullTTDUri);
+    typedef enum _JsTTDMoveModes : UINT64
+    {
+        /// <summary>
+        ///     Indicates no special actions needed for move.
+        /// </summary>
+        JsTTDMoveNone = 0x0,
 
-    /// <summary>
-    ///     TTD API -- may change in future versions:
-    ///     Ensure that the location specified for outputting the TTD data is clean. Specifically, ensure that any previous TTD
-    ///     in the location has been removed.
-    /// </summary>
-    /// <param name="fullTTDUri">The fully resolved location for the TTD data output as provied by JsTTDInitializeUriCallback.</param>
-    typedef void (CHAKRA_CALLBACK *JsTTDInitializeForWriteLogStreamCallback)(_In_z_ const wchar_t* uri);
+        /// <summary>
+        ///     Indicates that we want to move to the first/last event.
+        /// </summary>
+        JsTTDMoveFirstEvent = 0x1,
+        JsTTDMoveLastEvent = 0x2,
+
+        /// <summary>
+        ///     Indicates that we want to move to the kth event -- top 32 bits are event count.
+        /// </summary>
+        JsTTDMoveKthEvent = 0x4,
+
+        /// <summary>
+        ///     Indicates if we want to scan the snapshot interval containing the event to populate debug info before moving to execute event.
+        /// </summary>
+        JsTTDMoveScanIntervalBeforeDebugExecute = 0x10,
+
+        /// <summary>
+        ///     Indicates if we are doing the scan for a continue operation
+        /// </summary>
+        JsTTDMoveScanIntervalForContinue = 0x20,
+
+        /// <summary>
+        ///     Indicates if we want to set break on entry or just run and let something else trigger breakpoints.
+        /// </summary>
+        JsTTDMoveBreakOnEntry = 0x100
+    } JsTTDMoveMode;
 
     /// <summary>
-    ///     TTD API -- may change in future versions:
-    ///     Construct a HANDLE that will be used to read/write the event log portion of the TTD data based on the uri
-    ///     provided by JsTTDInitializeUriCallback.
+    ///     A handle for URI's that TTD information is written to/read from.
     /// </summary>
-    /// <remarks>
-    ///     <para>Exactly one of read or write will be set to true.</para>
-    /// </remarks>
-    /// <param name="uri">The fully resolved location for the TTD data as provied by JsTTDInitializeUriCallback.</param>
-    /// <param name="read">If the handle should be opened for reading.</param>
-    /// <param name="write">If the handle should be opened for writing.</param>
-    /// <returns>A HANDLE opened in read/write mode as specified.</returns>
-    typedef HANDLE (CHAKRA_CALLBACK *JsTTDGetLogStreamCallback)(_In_z_ const wchar_t* uri, _In_ bool read, _In_ bool write);
+    typedef void* JsTTDStreamHandle;
```

Comment:
You can use sth. similar to Windows DECLARE_HANDLE for a bit better type safety.